### PR TITLE
HP-2069: add staff role admin

### PR DESCRIPTION
**Pull request by bladeroot**
## Summary by CodeRabbit

- **New Features**
	- Introduced new roles: `role:server.staff-admin`, `role:hub.staff-admin`, and `role:staff-admin` for enhanced server and hub management capabilities.
	- Added new permissions including `charge.read`, `purse.set-credit`, and server-related permissions for improved access control.

- **Modifications**
	- Updated existing roles to clarify assignments, particularly for `role:server.admin`, `role:admin`, and `role:hub.admin`.
	- Reorganized permissions structure to better categorize responsibilities, including the reassignment of server and hub management permissions.

- **Tests**
	- Added new tests for the `role:staff-admin` to ensure proper access control.
	- Updated existing tests to reflect changes in role permissions and user assignments.

### CODE REVIEW

File: src/files/source/metadata.php [hunk 1]
```diff
@@ -1429,6 +1468,18 @@
...
         'description' => 'Allows reading of the server',
     ],
+    'server.read-all' => [
+        'description' => 'Allows reading of extended data of the server',
+    ],
+    'server.wizzard-read' => [
+        'description' => 'Allows reading info about wizzarding of the server',
+    ],
+    'server.read-legend' => [
+        'description' => 'Prohibits reading of the server',
```
Comment (SilverFire) [inline src/files/source/metadata.php:1478]: Тут треба поправити

File: src/files/source/metadata.php [hunk 2]
```diff
@@ -1429,6 +1468,18 @@
     'server.read' => [
         'description' => 'Allows reading of the server',
     ],
+    'server.read-all' => [
+        'description' => 'Allows reading of extended data of the server',
+    ],
```
Comment (SilverFire) [inline src/files/source/metadata.php:1473]: If I get a `server.read-all` permission, does it mean I also get `server.wizzard-read`, `server.read-legend`?

If yes – then `server.read-all` is a role, not a permission.

If no – then the name should be changed.

File: src/files/source/metadata.php [hunk 3]
```diff
@@ -1360,6 +1393,9 @@
     'purse.update' => [
         'description' => 'Allows updating of the purse',
     ],
+    'purse.st-credit' => [
```
Comment (SilverFire) [inline src/files/source/metadata.php:1396]:
```suggestion
    'purse.set-credit' => [
```

File: src/files/source/metadata.php [hunk 4]
```diff
@@ -1009,6 +1024,18 @@
     'deny:server.read' => [
         'description' => 'Prohibits reading of the server',
     ],
+    'deny:server.read-all' => [
+        'description' => 'Prohibits reading of extended data fo the server',
+    ],
+    'deny:server.read-billing' => [
+        'description' => 'Prohibits reading of tariff and sale information of server',
+    ],
+    'deny:server.wizzard-read' => [
```
Comment (SilverFire) [inline src/files/source/metadata.php:1033]:
```suggestion
    'deny:server.read-wizzard' => [
```

For other permissions, `read` is a prefix.

**Review by SilverFire — CHANGES_REQUESTED** [review]

File: src/files/source/metadata.php (hunk 2 above)
Comment (bladeroot) [reply src/files/source/metadata.php:1473]: No. Your proposition? This permission used in
hipanel-module-server:
    grid/ServerGridView.php
    grid/ServerRepresentations.php
    views/server/_search.php

File: src/files/source/tree.php [hunk 5]
```diff
@@ -35,28 +36,48 @@
...
     'role:server.manager' => [
-        'role:server.user', 'server.enable-block', 'server.disable-block', 'server.pay', 'server.sell', 'server.set-label',
+        'role:server.user',
+        'server.read-wizzard',
+        'server.enable-block', 'server.disable-block',
+        'server.pay', 'server.sell', 'server.set-label',
         'server.see-label',
+        'server.read-legend',
+        'server.read-all',
+        'server.read-manager',
```
Comment (hiqsol) [inline src/files/source/tree.php:54]: `manager` - це слово для ролі, а не для пермішена

File: src/files/source/metadata.php (hunk 2 above)
Comment (SilverFire) [reply src/files/source/metadata.php:1473]: - `server.read-financial-info`
- `server.read-system-info`

**Review by SilverFire — APPROVED** [review]

**Pull request merged**